### PR TITLE
AddressIPSocket: Fix EFAULT on invalid IP address presentation string

### PR DESCRIPTION
**Pull request by david-eckardt-sociomantic**
Make AddressIPSocket methods fail with `EAFNOSUPPORT` as their documentation specifies, not `EFAULT`.

IP address presentation strings (that is, strings that are expected to contain the address in the "a.b.c.d" notation) accepted by `AddressIPSocket` methods are forwarded to `IPAddress.opCall()`, which calls `inet_pton(3)` to parse them, and returns a `sockaddr*` pointer on success or null on failure, i.e. `inet_pton` failed. On failure the returned null pointer was passed to `connect(2)`, which then fails with `EFAULT` _Bad address_ (this would be a segfault in user space).

Ironically _Bad address_ is actually a helpful error message to report an invalid IP address.

### CODE REVIEW

**Review by nemanja-boric-sociomantic — APPROVED** [review]

**Comment by david-eckardt-sociomantic** [comment]
Updated with improved tests, found a bug in `InetAddress.opCall()` and fixed it.

File: src/ocean/sys/socket/AddressIPSocket.d [hunk 1]
```diff
@@ -393,3 +405,76 @@ class AddressIPSocket ( bool IPv6 = false ) : IPSocket!(IPv6), IAddressIPSocketI
         this.ip_address_len = this.in_address.inet_ntop(this.ip_address_).length;
     }
 }
+
+/// Test that methods that accept IP addresses in presentation notation
```
Comment (gavin-norman-sociomantic) [inline src/ocean/sys/socket/AddressIPSocket.d]: I believe using `///` here will make the following unittest be rendered as a usage example, which isn't what we want.

**Review by gavin-norman-sociomantic — COMMENTED** [review]
LGMT apart from one small point.

**Comment by david-eckardt-sociomantic** [comment]
Updated, fixing unit test comment.

**Pull request merged**